nanvm-lib/src/serializer/to_djs.rs

```diff
@@ -20,202 +30,158 @@ use super::to_json::WriteJson;
 /// Note that we use one ConstTracker for js objects and another for js arrays, keeping them
 /// separate - to reduce set sizes and save on operations.
 struct ConstTracker<D: Dealloc> {
-    visited_once: HashSet<Any<D>>,
-    visited_repeatedly: HashSet<Any<D>>,
+    visited: HashMap<Any<D>, Seen>,
 }
 
 impl<D: Dealloc> ConstTracker<D> {
     /// Returns true if `any` was visited before; updates the `const_tracker` set, tracking whether
     /// `any` was visited just once (it's in `const_tracker.visited_once`) or more than once (it's
     /// in `visited_repeatedly` in this case since we are up to writing it out as a const).
     fn is_visited(&mut self, any: &Any<D>) -> bool {
-        if self.visited_repeatedly.contains(any) {
-            // We've visited `any` more than once before, no action is needed here.
-            true
-        } else if self.visited_once.contains(any) {
-            // It's the second time we visit `any`, move it from `visited_once` to `to_do`.
-            self.visited_once.remove(any);
-            self.visited_repeatedly.insert(any.clone());
+        let optional_seen = self.visited.get_mut(any);
+        if let Some(seen) = optional_seen {
+            if *seen == Seen::Once {
+                *seen = Seen::Repeatedly;
+            }
```

Comment:
I think we don't need the `if`. We can always do `*seen = Seen::Repeatedly;`.

Reply:
Makes sense - I have corrected that in the next code review iteration.
Besides, I disabled a Clippy warning in an unrelated place via 
#[allow(clippy::to_string_trait_impl)]
(https://rust-lang.github.io/rust-clippy/master/index.html#/to_string_trait_impl)